ow_plexil/src/plexil-adapter/FaultHierarchy.h

```diff
@@ -0,0 +1,68 @@
+#ifndef Fault_Hierarchy
+#define Fault_Hierarchy
+
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+// Fault Hierarchy for use in the Interface class with fault lookups.
+
+struct Fault {
+  std::string name;
+  int status = 0;
+  std::vector<std::pair<std::string, std::string>> affected_subsystems;
+};
+
+struct Subsystem {
+  std::string name;
+  int hierarchy_faulted = 0;
+  int locally_faulted = 0;
+  int status = 0;
+  std::vector<std::string> faults;
+  std::unordered_map<std::string, int> severity_threshold = 
+    {
+      {"Low", 0},
+      {"Medium", 0},
+      {"High", 0}
+    };
+  std::unordered_map<std::string, int> current_severity =
+    {
+      {"Low", 0},
+      {"Medium", 0},
+      {"High", 0}
+    };
+  std::vector<std::pair<std::string, bool>> affected_subsystems;
+};
+
+class FaultHierarchy
+{
+public:
+
+  FaultHierarchy (std::string file_name, bool verbose_flag);
+  ~FaultHierarchy() = default;
+  FaultHierarchy (const FaultHierarchy&) = delete;
+  FaultHierarchy& operator= (const FaultHierarchy&) = delete;
+
+  // get and update functions
+  std::vector<bool> getSubsystemStatus(std::string name);
+  std::vector<std::string> getActiveFaults(std::string name);
```

Comment:
This could be const. Their definitions will need the same keyword in the cpp file. 
```suggestion
  std::vector<bool> getSubsystemStatus(std::string name) const;
  std::vector<std::string> getActiveFaults(std::string name) const;
```